Assets/Scripts/Augment/AugmentImplementations/DynamiteBarrel.cs

```diff
@@ -0,0 +1,137 @@
+using Mirror;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class DynamiteBarrel : GunBarrel
+{
+    private List<StickyDynamite> activeDynamites = new();
+    [SerializeField]
+    private StickyProjectileModifier stickyModifer;
+    [SerializeField]
+    private Mesh explodingBarrelMesh;
+    private bool isDetonatableEarly = false;
+    [SerializeField]
+    private MeshProjectileController meshProjectiles;
+    private SateliteUplink sateliteUplink;
+
+
+    void Start()
+    {
+        gunController = transform.parent.GetComponent<GunController>();
+        if (!gunController || !gunController.Player)
+            return;
+
+        gunController.Player.onDeath += OnDeath;
+        stickyModifer.OnStuckToTarget += AddDynamite;
+
+        // If bullets are large enough (enlarger), they should look like explosive barrels instead
+        if (Projectile.stats.ProjectileScale >= 3)
+            GetComponent<MeshProjectileController>().Vfx.SetMesh("Mesh", explodingBarrelMesh);
+
+        // If fire extension, dynamites should be able to explode while in air
+        if (gunController.GetComponentInChildren<Fire>())
+            isDetonatableEarly = true;
+        else
+            sateliteUplink = gunController.GetComponentInChildren<SateliteUplink>();
+
+        if (gunController.Player is AIManager)
+        {
+            StartCoroutine(TryDetonating());
+            return;
+        }
+        syncDirection = SyncDirection.ClientToServer;
+        gunController.Player.GetComponent<PlayerMovement>().ResetZoom();
```

Comment:
Please avoid calling ResetZoom etc without checking if this player has an input.... they may be network players

Reply:
We should really refactor the input mangager subscriptions sheningans at some point